pymc_experimental/statespace/filters/distributions.py

```diff
@@ -0,0 +1,271 @@
+        latent_obs_combined = _LinearGaussianStateSpace.dist(
+            a0, P0, c, d, T, Z, R, H, Q, steps=steps, **kwargs
+        )
+        k_states = T.type.shape[0]
+
+        latent_states = latent_obs_combined[..., :k_states]
+        obs_states = latent_obs_combined[..., k_states:]
+
+        return latent_states, obs_states
+
+
+class SequenceMvNormalRV(SymbolicRandomVariable):
+    default_output = 1
+    _print_name = ("SequenceMvNormal", "\\operatorname{SequenceMvNormal}")
+
+    def update(self, node: Node):
+        return {node.inputs[-1]: node.outputs[0]}
+
+
+class SequenceMvNormal(Distribution):
+    rv_op = SequenceMvNormalRV
+
+    def __new__(cls, *args, **kwargs):
+        support_shape = get_support_shape(
+            support_shape=None,
+            shape=None,
+            dims=kwargs.get("dims", None),
+            observed=None,
+            ndim_supp=2,
+        )
+
+        return super().__new__(cls, *args, support_shape=support_shape, **kwargs)
+
+    @classmethod
+    def dist(cls, mus, covs, logp, support_shape=None, **kwargs):
+        support_shape = get_support_shape(
+            support_shape=None,
+            shape=None,
+            dims=kwargs.get("dims", None),
+            observed=kwargs.get("observed", None),
+            ndim_supp=2,
+        )
+
+        if support_shape is None:
+            support_shape = pt.as_tensor_variable(())
+
+        steps = pm.intX(mus.shape[0])
+
+        return super().dist([mus, covs, logp, steps, support_shape], **kwargs)
+
+    @classmethod
+    def rv_op(cls, mus, covs, logp, steps, support_shape, size=None):
+        if size is not None:
+            batch_size = size
+        else:
+            batch_size = support_shape
+
+        mus_, covs_, support_shape_ = mus.type(), covs.type(), support_shape.type()
+        steps_ = steps.type()
+        logp_ = logp.type()
+
+        rng = pytensor.shared(np.random.default_rng())
+
+        def step(mu, cov, rng):
+            new_rng, mvn = pm.MvNormal.dist(
+                mu=mu, cov=stabilize(cov, JITTER_DEFAULT), rng=rng, size=batch_size
+            ).owner.outputs
+            return mvn, {rng: new_rng}
+
+        mvn_seq, updates = pytensor.scan(
+            step, sequences=[mus_, covs_], non_sequences=[rng], n_steps=steps_, strict=True
+        )
+
+        (seq_mvn_rng,) = tuple(updates.values())
+
+        mvn_seq_op = SequenceMvNormalRV(
+            inputs=[mus_, covs_, logp_, steps_], outputs=[seq_mvn_rng, mvn_seq], ndim_supp=2
+        )
+
+        mvn_seq = mvn_seq_op(mus, covs, logp, steps)
+        return mvn_seq
+
+
+@_logprob.register(SequenceMvNormalRV)
+def sequence_mvnormal_logp(op, values, mus, covs, logp, steps, rng, **kwargs):
+    # def step_logp(x, mu, cov):
+    #     return logp(pm.MvNormal.dist(mu, cov), x)
+    #
+    # logp_values, updates = pytensor.scan(
+    #     step_logp, sequences=[values[0], mus, covs], strict=True
+    # )
```

Comment:
This was an old attempt to compute the logp of the data separately, rather than providing it pre-computed from the filtering. I forgot to take it out.